http-auth/routes/routes_test.go

```diff
@@ -0,0 +1,166 @@
+package routes
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestMain(t *testing.T) {
+	t.Run("GET /", func(t *testing.T) {
+		request, _ := http.NewRequest(http.MethodGet, "/", nil)
+		rr := httptest.NewRecorder()
+
+		handler := http.HandlerFunc(IndexHandler)
+
+		handler.ServeHTTP(rr, request)
```

Comment:
What value does this construct bring over just calling `IndexHandler(rr, request)`?

(And throughout)

Reply:
Hmmm.. not sure. I look into it and doesn't really seem to do much.

I saw it being used for testing http and just went with it, but it makes sense to call the handler directly.